### PR TITLE
feat(analytics)connection time stats in join event

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Review by damencho — APPROVED** [review]

File: JitsiConference.js [hunk 1]
```diff
@@ -3670,7 +3670,26 @@ JitsiConference.prototype._sendConferenceJoinAnalyticsEvent = function() {
...
+    const connectionTimes = {
+        ...conferenceConnectionTimes,
+        ...xmppConnectionTimes,
+        ...globalNSConnectionTimes,
+        gumDuration: safeSubtract(gumEnd, gumStart),
+        xmppConnectingTime: safeSubtract(xmppConnectionTimes.connected, xmppConnectionTimes.connecting),
+        connectedToMUCJoinedTime: safeSubtract(
+            conferenceConnectionTimes['muc.joined'], xmppConnectionTimes.connected),
+        connectingToMUCJoinedTime: safeSubtract(
+            conferenceConnectionTimes['muc.joined'], xmppConnectionTimes.connecting),
```
Comment (bgrozev) [inline JitsiConference.js:3687]: This one too

File: JitsiConference.js [hunk 2]
```diff
@@ -3670,7 +3670,26 @@ JitsiConference.prototype._sendConferenceJoinAnalyticsEvent = function() {
...
+    const gumEnd = window.connectionTimes['firstObtainPermissions.end'];
+    const globalNSConnectionTimes = window.JitsiMeetJS?.app?.connectionTimes ?? {};
+    const connectionTimes = {
+        ...conferenceConnectionTimes,
+        ...xmppConnectionTimes,
+        ...globalNSConnectionTimes,
+        gumDuration: safeSubtract(gumEnd, gumStart),
+        xmppConnectingTime: safeSubtract(xmppConnectionTimes.connected, xmppConnectionTimes.connecting),
+        connectedToMUCJoinedTime: safeSubtract(
+            conferenceConnectionTimes['muc.joined'], xmppConnectionTimes.connected),
```
[the remark this answers is not in the document]
Comment (hristoterezov) [reply JitsiConference.js:3685]: Hmm what do you mean? muc joined is after xmpp connected, so we should do muc.joined - connected, isn't it?

File: JitsiConference.js (hunk 1 above)
Comment (hristoterezov) [reply JitsiConference.js:3687]: Hmm what do you mean? muc joined is after xmpp connected, so we should do muc.joined - connected, isn't it?

File: JitsiConference.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgrozev) [reply JitsiConference.js:3685]: Yep, I'm an idiot :)

**Pull request merged**